pylint/checkers/variables.py

```diff
@@ -607,6 +609,25 @@ def get_next_to_consume(self, node):
                 or n.statement().parent_of(node)
             ]
 
+        # If this node is in a Finally block of a Try/Finally,
+        # filter out assignments in the try portion, assuming they may fail
+        if (
+            found_nodes
+            and isinstance(node.statement().parent, nodes.TryFinally)
+            and node.statement() in node.statement().parent.finalbody
+        ):
+            filtered_nodes = [
+                n
+                for n in found_nodes
+                if not (
+                    isinstance(n.statement().parent, nodes.TryFinally)
+                    and n.statement() in n.statement().parent.body
+                )
+            ]
+            difference = set(found_nodes) - set(filtered_nodes)
+            self.mark_as_consumed(name, difference)
+            found_nodes = filtered_nodes
```

Comment:
Sorry for the additional push after approval, but I think this is safer. I noticed while working on the similar issue #4761.